### PR TITLE
add support for meta-data

### PR DESCRIPTION
**Pull request by shabtaisharon**

### CODE REVIEW

File: Ds3/Helpers/Jobs/Job.cs [hunk 1]
```diff
@@ -65,6 +67,12 @@ public TSelf WithCancellationToken(CancellationToken cancellationToken)
             return (TSelf)(IBaseJob<TSelf, TItem>)this;
         }
 
+        public TSelf WithMetadata(IMetadataAccess metadataAccess)
+        {
+            this._metadataAccess = metadataAccess;
```
Comment (rpmoore) [inline Ds3/Helpers/Jobs/Job.cs]: We should add a check to make sure that if this is called after the Transfer method is called, an exception is thrown.

Comment (shabtaisharon) [inline Ds3/Helpers/Jobs/Job.cs]: i've add the exception for all 3 with* function that have the same restrictions

File: Ds3/Helpers/Jobs/Job.cs [hunk 2]
```diff
@@ -45,26 +45,41 @@ internal abstract class Job<TSelf, TItem> : IBaseJob<TSelf, TItem>
         private readonly IDs3Client _client;
         private readonly MasterObjectList _jobResponse;
         private Func<TItem, Stream> _createStreamForTransferItem;
+        private IMetadataAccess _metadataAccess;
+        private bool TransferStarted { get; set; }
```
Comment (rpmoore) [inline Ds3/Helpers/Jobs/Job.cs]: Why did you make this a private property rather than a private variable?

Comment (shabtaisharon) [inline Ds3/Helpers/Jobs/Job.cs]: no resone, just convenience

Comment (rpmoore) [inline Ds3/Helpers/Jobs/Job.cs]: does it default to false?

Comment (shabtaisharon) [inline Ds3/Helpers/Jobs/Job.cs]: yes

**Pull request merged**